src/components/sbb-form-field/sbb-form-field.tsx

```diff
@@ -225,15 +224,37 @@ export class SbbFormField implements ComponentInterface {
       attributes: true,
       attributeFilter: ['readonly', 'disabled', 'class'],
     });
+    this._element.dataset.inputType = this._input.tagName.toLowerCase();
+    this._syncLabelInputReferences();
+  }
 
-    if (!this._input.id) {
-      this._input.id = `sbb-form-field-input-${nextId++}`;
+  private _syncLabelInputReferences(): void {
+    if (!this._input || !this._labels.length) {
+      return;
     }
 
-    this._element.dataset.inputType = this._input.tagName.toLowerCase();
-    const label = this._element.querySelector('label');
-    if (label) {
-      label.htmlFor = this._input.id;
+    if (this._supportedNativeInputElements.includes(this._input.tagName)) {
```

Comment:
I think there would be some more branches and specialities which could be tested

Reply:
Refactor to only use one label and document accordingly